src/admin/hooks/useRoles.ts

```diff
@@ -0,0 +1,40 @@
+import { useState, useEffect } from 'react';
+
+import { useAuthentication } from '../../authentication';
+import { Role, fetchRoles } from '../../api';
+
+type NullableError = Error | null;
+
+export default function useRoles(): {
+  loading: boolean;
+  error: NullableError;
+  roles: Role[];
+} {
+  const { token } = useAuthentication();
+
+  const [loading, setLoading] = useState(false);
+  const [error, setError] = useState(null as NullableError);
+  const [roles, setRoles] = useState([] as Role[]);
+
+  useEffect(() => {
```

Comment:
Once https://github.com/cov-clear/web/pull/25 is merged, I'll rebase and make use of the common `useAuthenticatedHttpResource` to reduce duplication.

Reply:
Was going to add a comment about cancellation but that'll take care of it i guess